File: migrate_blogs.py
import json
import urllib.request
import urllib.error
import time
import ssl

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE
import urllib.error
import time

API_URL = 'https://www.studym.co.kr/api/blog/'
ADMIN_SECRET = 'studym001!'
JSON_FILE = 'public/content/articles.json'

def migrate():
    with open(JSON_FILE, 'r', encoding='utf-8') as f:
        articles = json.load(f)

    print(f"Found {len(articles)} legacy articles to migrate.")

    for article in articles:
        payload = {
            "title": article.get("title"),
            "slug": article.get("slug"),
            "category": article.get("category"),
            "excerpt": article.get("excerpt"),
            "content": article.get("content"),
            "author": article.get("author", "Study M 교육연구소"),
            "read_time": article.get("readTime"),
            "tags": article.get("tags", []),
            "thumbnail": article.get("thumbnail"),
            "date": article.get("date")
        }

        req = urllib.request.Request(API_URL, method='POST')
        req.add_header('Content-Type', 'application/json')
        req.add_header('x-admin-secret', ADMIN_SECRET)
        data = json.dumps(payload).encode('utf-8')

        print(f"Migrating \"{payload['title']}\"... ", end='', flush=True)

        try:
            with urllib.request.urlopen(req, data=data, context=ctx) as response:
                res_data = json.loads(response.read().decode('utf-8'))
                print(f"Success (ID: {res_data.get('data', {}).get('id')})")
        except urllib.error.HTTPError as e:
            err_text = e.read().decode('utf-8')
            try:
                err_data = json.loads(err_text)
                if err_data.get("error") == "이미 존재하는 URL 슬러그입니다.":
                    print("Skipped (Already exists)")
                else:
                    print(f"Failed: {err_data}")
            except Exception:
                print(f"Failed with status {e.code}: {err_text}")
        except Exception as e:
            print(f"Error: {e}")

        time.sleep(0.5)

    print("Migration Complete!")

if __name__ == '__main__':
    migrate()

File: migrate_blogs.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const API_URL = 'https://study-manager-production-826b.up.railway.app/api/blog/';
const ADMIN_SECRET = 'studym001!';

async function migrate() {
    try {
        const dataPath = path.join(__dirname, 'public', 'content', 'articles.json');
        const rawData = fs.readFileSync(dataPath, 'utf8');
        const articles = JSON.parse(rawData);

        console.log(`Found ${articles.length} legacy articles to migrate.`);

        for (const article of articles) {
            const payload = {
                title: article.title,
                slug: article.slug,
                category: article.category,
                excerpt: article.excerpt,
                content: article.content,
                author: article.author || 'Study M 교육연구소',
                read_time: article.readTime,
                tags: article.tags,
                thumbnail: article.thumbnail,
                date: article.date
            };

            process.stdout.write(`Migrating "${payload.title}"... `);
            const response = await fetch(API_URL, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'x-admin-secret': ADMIN_SECRET
                },
                body: JSON.stringify(payload)
            });

            if (response.ok) {
                const resData = await response.json();
                console.log(`Success (ID: ${resData.data.id})`);
            } else {
                const errText = await response.text();
                try {
                    const errData = JSON.parse(errText);
                    if (errData.error === '이미 존재하는 URL 슬러그입니다.') {
                        console.log(`Skipped (Already exists)`);
                    } else {
                        console.log(`Failed:`, errData);
                    }
                } catch (parseError) {
                    console.log(`Failed with status ${response.status}: ${errText}`);
                }
            }
            // Add a small delay
            await new Promise(resolve => setTimeout(resolve, 500));
        }
        console.log('Migration Complete!');
    } catch (e) {
        console.error('\nFatal Error Migration:', e);
    }
}

migrate();
